snuba/snuba_migrations/search_issues/0001_search_issues.py

```diff
@@ -0,0 +1,102 @@
+from typing import List, Sequence
+
+from snuba.clickhouse.columns import (
+    UUID,
+    Array,
+    Column,
+    DateTime,
+    IPv4,
+    IPv6,
+    Nested,
+    String,
+    UInt,
+)
+from snuba.clusters.storage_sets import StorageSetKey
+from snuba.migrations import migration, operations, table_engines
+from snuba.migrations.columns import MigrationModifiers as Modifiers
+from snuba.migrations.operations import OperationTarget, SqlOperation
+
+columns: List[Column[Modifiers]] = [
+    Column("organization_id", UInt(64)),
+    Column("project_id", UInt(64)),
+    Column("group_id", UInt(64)),
+    Column("search_title", String()),
+    Column("primary_hash", UUID()),
+    Column("fingerprint", Array(String())),
+    Column("occurrence_id", UUID()),
+    Column("occurrence_type_id", UInt(8)),
+    Column("detection_timestamp", DateTime()),
+    Column("timestamp", DateTime(Modifiers(materialized="receive_timestamp"))),
```

Comment:
Curious why you needed this materialization? Isn't this kind of a waste to keep 2 copies of the same thing?

Reply:
This was more of an assumption on my part on how snuba_sdk builds and enforces queries with the time column.
https://github.com/getsentry/snuba-sdk/blob/8b8fd04a56f8afb3ca9909ad26b937b595360e2d/snuba_sdk/entity.py#L65-L66

I was thinking of adding a default value of `timestamp` there so any new datasets spun up would have a reasonable default for a time column. 

Unfortunately, our search code is fairly old and there's a lot of .. accumulated logic there that we need to keep for the short term.

Reply:
I think `timestamp` should just be the value of `client_timestamp` tbh. I'm pretty sure that's what it is in the errors table. I think I was also confused before, and the `order_by` should be basically the same as errors:

`order_by="(project_id, toStartOfDay(timestamp), primary_hash, cityHash64(event_id))`

Reply:
Snuba (and snuba sdk) does enforce a time column, but the time column can be anything you want, it doesn't have to be called `timestamp`. If you absolutely must have the other column, how about making it an expression that just points to `receive_timestamp` instead of a materialization that keeps another copy of the data?

Reply:
The comment about the order_by change is a blocker if you want to support replacements in the future. Once an order by field is setup on a table, you cannot change it.

Reply:
We're ok with rebuilding the table from scratch short term, but we may as well fix this now since we're aware of it.